pkgs/stdenv/adapters.nix

```diff
@@ -245,6 +246,11 @@ rec {
          * wrapper instead? */
         cc = targetStdenv.cc.cc;
 
+        /* NOTE: We must provide coreutils explicitly so that wrapCCWith
+         * doesn't take it from pkgs, which is pkgsHostTarget, because it
+         * would fail to run on the buildPlatform when cross-compiling. */
+        coreutils = targetStdenv.cc.coreutils_bin;
```

Comment:
Should we do this, or should we add another passthru? I'm uncomfortable with relying on `optionalString` not enforcing types

Reply:
I think it's fine, pending a larger rewrite/refactoring of the whole interface (as you've noted in #282356 and #283517. Is that agreeable to you? I'd like to get this merged.